packages/cubejs-schema-compiler/src/adapter/ClickHouseQuery.ts

```diff
@@ -281,6 +338,22 @@ export class ClickHouseQuery extends BaseQuery {
     // ClickHouse intervals have a distinct type for each granularity
     delete templates.types.interval;
     delete templates.types.binary;
+
+    const collation = this.getCollation();
+
+    if (collation) {
+      templates.expressions.sort = `${templates.expressions.sort}{% if data_type and data_type == 'string' %} COLLATE '${collation}'{% endif %}`;
+      templates.expressions.order_by = `${templates.expressions.order_by}{% if data_type and data_type == 'string' %} COLLATE '${collation}'{% endif %}`;
+
+      const oldOrderBy = '{% if order_by %}\nORDER BY {{ order_by | map(attribute=\'expr\') | join(\', \') }}{% endif %}';
+
+      const newOrderBy =
+        '{% if order_by %}\nORDER BY {% for item in order_by %}{{ item.expr }}' +
+        `{%- if item.data_type and item.data_type == 'string' %} COLLATE '${collation}'{% endif %}` +
+        '{%- if not loop.last %}, {% endif %}{% endfor %}{% endif %}';
```

Comment:
I dislike the idea of replacing strings by putting the original `ORDER BY` expression here. The `statements/select` template might change over time in BaseQuery, and if `ORDER BY` condition is touched, this code would break.

I would suggest moving **whole** `statements/select` template here, then deciding the `ORDER BY` part based on the provided condition. This way if BaseQuery' `statements/select` is modified, this dialect would be untouched and would continue working regardless. We already use similar approach with overriding whole `statements/select` template in `PinotQuery` and `PrestodbQuery`.